marklogic-data-hub/src/test/java/com/marklogic/hub/flow/impl/StemmingTest.java

```diff
@@ -0,0 +1,102 @@
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.flow.RunFlowResponse;
+import com.marklogic.hub.impl.Versions;
+import com.marklogic.hub.job.JobStatus;
+import com.marklogic.mgmt.api.API;
+import com.marklogic.mgmt.api.database.Database;
+import com.marklogic.mgmt.resource.databases.DatabaseManager;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class StemmingTest extends AbstractHubCoreTest {
+
+    Database db;
+    @Autowired
+    FlowRunnerImpl flowRunner;
+
+    @BeforeEach
+    public void setupEach() {
+        HubConfig hubConfig = runAsAdmin();
+        db = new Database(new API(hubConfig.getManageClient()), hubConfig.getDbName(DatabaseKind.STAGING));
+        db.setStemmedSearches("advanced");
+        db.save();
+        installProjectInFolder("flow-runner-test");
+    }
+    /*
+     https://bugtrack.marklogic.com/54003
+     When stemmed search is set to "advanced", fetching legacy mappings (and/or step definition) fails and hence running
+     testFlow would fail. After that if db is reindexed, fetching of artifacts succeeds. In order to apply the fix for
+     bug 54003 in server, "Reindex JSON language property” trace event has to be added and the test should be run on
+     nightly servers. With the fix applied, the flow will fail after reindexing as well as queries requires 'lang=zxx'
+     to be included so as to fetch artifacts.
+     */
+    @Test
+    void runFlowWithStemmingSearch() {
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assumptions.assumeTrue(mlVersion.getMajor() >=10);
+
+        try{
+            RunFlowResponse resp = runFlow();
+            Assertions.assertTrue(JobStatus.STOP_ON_ERROR.toString().equalsIgnoreCase(resp.getJobStatus()));
+
+            HubConfig hubConfig = runAsAdmin();
+            DatabaseManager dbManager = new DatabaseManager(hubConfig.getManageClient());
+            dbManager.reindexDatabase(hubConfig.getDbName(DatabaseKind.STAGING));
+
+            waitForReIndex(runAsAdmin());
+            resp = runFlow();
+            Assertions.assertTrue(JobStatus.FINISHED.toString().equalsIgnoreCase(resp.getJobStatus()));
+
+        }
+        finally {
+            db.setStemmedSearches("off");
+            db.save();
+        }
+    }
+
+    private RunFlowResponse runFlow(){
+        runAsDataHubOperator();
+        RunFlowResponse resp = null;
+        try{
+            resp = flowRunner.runFlow("testFlow", null, null, null, null);
+        }
+        catch (Exception e){
+            logger.error("Flow run failed;cause: "+ e.getMessage());
+        }
+
+        flowRunner.awaitCompletion();
+        return resp;
+    }
+
+    private void waitForReIndex(HubConfig hubConfig){
```

Comment:
If this is reliable, I think it's a good thing to move up to AbstractHubTest. But for consistency, do "Reindex" instead of "ReIndex", as ML treats it as one word. 

I'm also going to open a ticket for this in ml-app-deployer, this would be a nice feature to have in there - e.g. new Database(...).waitForReindex().